py/uninameslist.py

```diff
@@ -86,19 +99,11 @@ def name(char):
 
 def name2(char):
     '''returns the normative alias if defined for correcting a Unicode character name, else just the name'''
-    # NOTE: As of Unicode 10.0 these are only the 25 characters:
-    #   U+01A2 U+01A3 U+0709 U+0CDE U+0E9D U+0E9F U+0EA3 U+0EA5 U+0FD0 U+11EC U+11ED U+11EE U+11EF
-    #   U+2118 U+2448 U+2449 U+2B7A U+2B7C U+A015 U+FE18 U+FEFF U+0122D4 U+0122D5 U+01B001 U+01D0C5
-    # ... but the function is provided if you want it.
-    annot = annotation(char)
     try:
-        # NOTE: index should not be needed as NamesList.txt always defines the normative alias first, but just keeping safe
-        correctedNamePos = annot.index("\t% ") + 3
-        try:
-            nextTabPos = correctedNamePos + annot[correctedNamePos:].index("\n\t")  # there may be annotations beyond the alias
-            return annot[correctedNamePos:nextTabPos]
-        except ValueError:
-            return annot[correctedNamePos:]
+        name2Index = _codepointsWithName2.index(ord(char))
+        annotationBytes = _lib.uniNamesList_names2anC(name2Index)
+        normativeAliasLength = _lib.uniNamesList_names2lnC(name2Index)
+        return annotationBytes[:normativeAliasLength].decode()
     except ValueError:
         return name(char)
 
```

Comment:
I think this might be faster (searching in C, we're maybe using 1000+ CPU cycles to check if char is one of 25 codes, but searching a python list, we're maybe running 10000 to 100000+ CPU cycles). Best way to test, is to start a timer, search for 1000 codes, stop timer, and see how long it takes.
```C
def name2(char):
    name2Index = _lib.uniNamesList_names2lnU(ord(char))
    if name2Index < 0 then return name(char)
    annotationBytes = _lib.uniNamesList_names2anC(name2Index)
    normativeAliasLength = _lib.uniNamesList_names2lnC(name2Index)
    return annotationBytes[:normativeAliasLength].decode()
```
I did not test this code above, but it's approximately close, and might be worth testing if it helps speed up the routine from a python point of view.

Reply:
At first, I did not think it was useful, but I've added a patch to make uniNamesList_names2getU() visible. This should make the above function work (avoid having to find name2index repeatedly).